src/vectors/vectorWindow.component.tsx

```diff
@@ -41,23 +54,46 @@ const VectorWindow = (props: VectorWindowProps) => {
     limit: vector?.data.length || 0,
   });
 
-  const slicedVector: Vector = {
-    data: (vector || { data: [] }).data.slice(range.skip, range.limit),
-  };
-  const slicedLabels: string[] = (
-    labels ||
-    Array.from({ length: vector?.data.length || 0 }, (_, i) => i.toString())
-  ).slice(range.skip, range.limit);
+  React.useEffect(() => {
+    const userPrefVector =
+      vectorLimitData === null ? NaN : Number(vectorLimitData);
+
+    const newVectorLimit = isNaN(userPrefVector)
+      ? labels?.length
+      : Math.min(userPrefVector, labels?.length || 0);
+
+    setRange((prev) => ({ ...prev, limit: newVectorLimit || 0 }));
+  }, [labels, vectorLimitData]);
+
+  React.useEffect(() => {
+    const userPrefVector =
+      vectorSkipData === null ? NaN : Number(vectorSkipData);
+
+    const newVectorSkip = isNaN(userPrefVector)
+      ? 0
+      : Math.min(userPrefVector, labels?.length || 0);
+
+    setRange((prev) => ({ ...prev, skip: newVectorSkip }));
+  }, [labels?.length, vectorSkipData]);
```

Comment:
One of these `useEffects` is using `labels?.length` in it's dependency array and the other is using `labels` - should probably unify. If `labels?.length` works we should probably use that once since it's probably more efficient.